packages/browser/src/sdk.ts

```diff
@@ -60,22 +60,32 @@ function applyDefaultOptions(optionsArg: BrowserOptions = {}): BrowserOptions {
   return { ...defaultOptions, ...optionsArg };
 }
 
+type ExtensionProperties = {
+  chrome?: Runtime;
+  browser?: Runtime;
+};
+type Runtime = {
+  runtime?: {
+    id?: string;
+  };
+};
+
 function shouldShowBrowserExtensionError(): boolean {
-  const windowWithMaybeChrome = WINDOW as typeof WINDOW & { chrome?: { runtime?: { id?: string } } };
-  const isInsideChromeExtension =
-    windowWithMaybeChrome &&
-    windowWithMaybeChrome.chrome &&
-    windowWithMaybeChrome.chrome.runtime &&
-    windowWithMaybeChrome.chrome.runtime.id;
-
-  const windowWithMaybeBrowser = WINDOW as typeof WINDOW & { browser?: { runtime?: { id?: string } } };
-  const isInsideBrowserExtension =
-    windowWithMaybeBrowser &&
-    windowWithMaybeBrowser.browser &&
-    windowWithMaybeBrowser.browser.runtime &&
-    windowWithMaybeBrowser.browser.runtime.id;
-
-  return !!isInsideBrowserExtension || !!isInsideChromeExtension;
+  const windowWithMaybeExtension = WINDOW as typeof WINDOW & ExtensionProperties;
+
+  const extensionKey = windowWithMaybeExtension.chrome ? 'chrome' : 'browser';
```

Comment:
to clarify, when would you have `window.browser.xxx`, and would that mean this is an extension??

Reply:
Yes, we introduced this in the original browser extension check (https://github.com/getsentry/sentry-javascript/pull/10844/files)